packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -101,18 +99,33 @@ const isAllowedL2 = async ({
   )
 }
 
-function useTokenFromSearchParams(): string | undefined {
-  const [{ token: tokenFromSearchParams }] = useArbQueryParams()
+export function useTokenFromSearchParams(): {
+  tokenFromSearchParams: string | undefined
+  setTokenQueryParam: (token: string | undefined) => void
+} {
+  const [{ token: tokenFromSearchParams }, setQueryParams] = useArbQueryParams()
+
+  const setTokenQueryParam = (token: string | undefined) =>
+    setQueryParams({ token })
 
   if (!tokenFromSearchParams) {
-    return undefined
+    return {
+      tokenFromSearchParams: undefined,
+      setTokenQueryParam
+    }
   }
 
   if (!isAddress(tokenFromSearchParams)) {
-    return undefined
+    return {
+      tokenFromSearchParams,
```

Comment:
Is it okay it return `tokenFromSearchParams` if it's an invalid address? We returned `undefined` before this change.

If it's okay, we can simply remove this check and keep:
```
return {
  tokenFromSearchParams,
  setTokenQueryParam
}
```

As they return the same data.

Reply:
<img width="519" alt="image" src="https://github.com/OffchainLabs/arbitrum-token-bridge/assets/13184582/8432bff4-09c1-44c7-9cf5-704c1818d92d">
we need a check here because ethers' `isAddress` only accepts string but it could be `null` / `undefined`

Reply:
We can remove the check for `isAddress` and return the default state on L125 directly

Reply:
BTW did you check about rerender? Is it returning a new reference of the function on every render?